### ISSUE TITLE
False negatives when passing `subprocess` command as keyword argument

### ISSUE THREAD

**Issue by allanlewis** [astral-sh/ruff#27631]
### Summary

Various [`flake8-bandit`-derived rules](https://docs.astral.sh/ruff/rules/#flake8-bandit-s) are triggered when calling one of the `subprocess` functions with a non-literal value as the first argument, as this is considered untrusted input. However, passing the array of command tokens as a keyword argument, i.e. `args=[...]`, silences the violation!

```console
➜ ruff check --select=S603 - <<< 'subprocess.check_output([foo])'
All checks passed!
➜ ruff check --select=S603 - <<< 'import subprocess; foo = input(); subprocess.check_output([foo])'
S603 `subprocess` call: check for execution of untrusted input
 --> -:1:35
  |
1 | import subprocess; foo = input(); subprocess.check_output([foo])
  |                                   ^^^^^^^^^^^^^^^^^^^^^^^

Found 1 error.
```

This appears to be due to the use of `call.arguments.args.first()` in this file:
https://github.com/astral-sh/ruff/blob/5b48a040974781ba90b47c8df628f8fd9b6c95dd/crates/ruff_linter/src/rules/flake8_bandit/rules/shell_injection.rs#L321-L327

### Version

ruff 0.16.2 (5b48a0409 2026-08-07)

**Comment by ntBre** [comment]
Thanks for the report. I think in this case, I'd actually consider this a false negative on the keyword argument rather than a false positive on the positional argument since these rules are intentionally aggressive about flagging potential security issues.

We should use one of our argument helpers here instead of `args.first()`, as you said:

https://github.com/astral-sh/ruff/blob/9a908ef2a99b00056e465dc7aec55f69304b64c0/crates/ruff_python_ast/src/nodes.rs#L3574

**Comment by allanlewis** [comment]
You're absolutely right, I meant false negatives 🙄
